migration/idempotent/001-base.sql

```diff
@@ -1447,6 +1458,34 @@ COMMENT ON FUNCTION prom_api.set_metric_chunk_interval(TEXT, INTERVAL)
 IS 'set a chunk interval for a specific metric (this overrides the default)';
 GRANT EXECUTE ON FUNCTION prom_api.set_metric_chunk_interval(TEXT, INTERVAL) TO prom_admin;
 
+CREATE OR REPLACE FUNCTION prom_api.get_metric_chunk_interval(metric_name TEXT)
+    RETURNS INTERVAL
+    SET search_path = pg_catalog, pg_temp
+AS $func$
+    DECLARE
+    _table_name TEXT;
+    _is_default BOOLEAN;
+    _chunk_interval INTERVAL;
+    BEGIN
+        SELECT table_name, default_chunk_interval
+        INTO _table_name, _is_default
+        FROM _prom_catalog.metric WHERE table_schema = 'prom_data' AND metric.metric_name = get_metric_chunk_interval.metric_name;
+        IF _is_default THEN
+            RETURN prom_api.get_default_chunk_interval();
+        END IF;
+        SELECT _timescaledb_internal.to_interval(d.interval_length)
+        INTO _chunk_interval
+        FROM _timescaledb_catalog.hypertable h
+            JOIN _timescaledb_catalog.dimension d on d.hypertable_id = h.id
```

Comment:
There's no way a metric hypertable can have more than the one dimension on time, correct?

Reply:
Good question. I'm pretty sure not. It's not so clear from this query, but I've rewritten this to only use the (public) `timescaledb_information.dimensions` table, which I think makes it more obvious.